numba/unicode.py

```diff
@@ -1361,6 +1361,25 @@ def impl(a):
     return impl
 
 
+# https://github.com/python/cpython/blob/1d4b6ba19466aba0eb91c4ba01ba509acf18c723/Objects/unicodeobject.c#L12056-L12085    # noqa: E501
+@overload_method(types.UnicodeType, 'isdigit')
+def unicode_isdigit(data):
+    """Implements UnicodeType.isdigit()"""
+
+    def impl(data):
+        length = len(data)
```

Comment:
Any merit in implementing this https://github.com/python/cpython/blob/1d4b6ba19466aba0eb91c4ba01ba509acf18c723/Objects/unicodeobject.c#L12070-L12074

I'm starting to think that these "single character string" branches are actually optimisations for the case that's most encountered in practice. I think the CPython impls deals with the cases in order of frequency of occurrence mixed with cost, i.e. 1 character (most likely), then empty string (cheap), then general case.